src/ResourceManager/Kusto/Commands.Kusto/Commands/TestAzureRmKustoClusterName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Kusto.Models;
+using Microsoft.Azure.Commands.Kusto.Utilities;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Kusto.Properties;
+using Microsoft.Azure.Management.Kusto.Models;
+
+namespace Microsoft.Azure.Commands.Kusto
+{
+    [Cmdlet("Test", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "KustoClusterName", DefaultParameterSetName = ParameterSet),
+        OutputType(typeof(PSKustoClusterNameAvailability))]
+    public class TestAzureRmKustoClusterName : KustoCmdletBase
+    {
+        protected const string ParameterSet = "ByClusterOrResourceGroupOrSubscription";
+
+        [Parameter(
+            ParameterSetName = ParameterSet,
+            Mandatory = false,
+            HelpMessage = "The location where to check.")]
+        public string Location { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParameterSet,
+            Mandatory = false,
+            HelpMessage = "Name of a specific cluster.")]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string location = Location;
+            string clusterName = Name;
+            if (string.IsNullOrEmpty(location))
```

Comment:
I don't understand these two if checks... you haven't made a call to the server so how do you know if the cluster exists?